nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfigEdit.js

```diff
@@ -808,7 +809,21 @@ export function RanEdit(props: Props) {
       setError(e.response?.data?.message ?? e.message);
     }
   };
-
+  const isEnodebUnregistered = React.useCallback(
```

Comment:
@MightyDani Basic question why is this a callback and not a simple function ?

Reply:
Andrei asked me to call this function only when the state changed instead of calling it every time. You can see his comments on the original peer request I made.

Reply:
Sorry, I didn't explain it the first time well. The useCallback helps us do memoization. So, the first time it is called, it does all the checking and renders it. However, after the first call, when we call it second time or any number of times, since it already memoized it, it knows and returns the values as long as the the state didn't change. Does that makes sense?

Reply:
Just to give you some option, I did the useEffect(just as what you suggested) at the start and gave it a dependency of gateway id and the connected enodeb serials because if any of the 2 changed we need to redo it again. I tested it and the functionality worked. If you don't want to go with the memoization technique just reply to me and I will push the new changes I made.  
<img width="530" alt="Screen Shot 2021-06-04 at 11 54 47 AM" src="https://user-images.githubusercontent.com/34602743/120829812-d3a4f180-c52b-11eb-994a-8068d623df44.png">
<img width="593" alt="Screen Shot 2021-06-04 at 11 55 26 AM" src="https://user-images.githubusercontent.com/34602743/120829823-d99ad280-c52b-11eb-94bc-52817c9cd216.png">
[git_diff.txt](https://github.com/magma/magma/files/6599742/git_diff.txt)

Reply:
I also did the suggestions you made and wrote a useEffect function which has a dependency list containing the gateway id and the connected enodeb serials for that gateway id since the useEffect needs to run again whenever one of the two things changes. If you would want to go with the useEffect method instead of the memoization technique, just call me and I will push the changes.
<img width="530" alt="Screen Shot 2021-06-04 at 11 54 47 AM" src="https://user-images.githubusercontent.com/34602743/120834562-3056db00-c531-11eb-8ece-3146a169a1ad.png">
<img width="593" alt="Screen Shot 2021-06-04 at 11 55 26 AM" src="https://user-images.githubusercontent.com/34602743/120834585-33ea6200-c531-11eb-8c85-2d0ec607b5b8.png">
[git_diff.txt](https://github.com/magma/magma/files/6599969/git_diff.txt)